### PR TITLE
typst-preview: init at 0.9.0

### PR DESCRIPTION
**Pull request by berberman**
## Description of changes

This PR adds the binary part of `typst-preview` as a package. It can be used as a typst preview server without vscode.

Also, notice that currently we are patching the binary bundled in vscode extension `typst-preview`to make things work:

https://github.com/NixOS/nixpkgs/blob/7eddbf56907e20b14a2996597cfc6619d1c18aed/pkgs/applications/editors/vscode/extensions/default.nix#L2300-L2339

IMO it would be better to leverage this in the vscode extension somehow // cc @drupol

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by drupol** [comment]
Eval failure?

**Comment by berberman** [comment]
That was caused by IFD in `mkYarnPackage`... Fixed now

**Comment by drupol** [comment]
I think it would be super easy to fix the issue on Darwin. Do you think you might have a look?

**Comment by berberman** [comment]
Sorry, what issue?

**Review by drupol — APPROVED** [review]

**Comment by drupol** [comment]
excellent, thanks!

File: pkgs/tools/typesetting/typst-preview/default.nix [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  };
+
+  prePatch = ''
+    mkdir -p addons/vscode/out/frontend
+    cp -R ${frontend}/* addons/vscode/out/frontend/
+  '';
+
+  meta = with lib; {
+    description = "Preview your Typst files in vscode";
+    homepage = "https://github.com/Enter-tainer/typse-preview";
```
Comment (drupol) [inline pkgs/tools/typesetting/typst-preview/default.nix:68]: I just noticed this typo :(

**Pull request merged**